cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java

```diff
@@ -692,11 +696,202 @@ private void handlePostSubscription(HttpServletRequest request, HttpServletRespo
                 response.getWriter().println("{\"success\":\"false\","
                             + "\"result\" : {" + orionJson.toJSONString() + "}");
             } // if else
+            
         } catch (Exception e) {
             LOGGER.error(e.getMessage());
         } // try catch
+        
     } // handlePostSubscription
     
+    private void handleDeleteSubscription(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("json;charset=utf-8");
+        
+        String subscriptionId = request.getParameter("subscription_id");
+        String ngsiVersion = request.getParameter("ngsi_version");
+        
+        if ((subscriptionId == null) || (subscriptionId.equals(""))) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            response.getWriter().println("{\"success\":\"false\","
+                    + "\"error\":\"Parse error, wrong parameter (subscription_id). Check it for errors.\"");
+            LOGGER.error("Parse error, wrong parameter (subscription_id). Check it for errors.\"}");
+            return;
+        } // if
+        
+        if ((ngsiVersion == null) || (ngsiVersion.equals(""))) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            response.getWriter().println("{\"success\":\"false\","
+                    + "\"error\":\"Parse error, wrong parameter (ngsi_version). Check it for errors.\"");
+            LOGGER.error("Parse error, wrong parameter (ngsi_version). Check it for errors.\"}");
+            return;
+        } // if
+                
+        LOGGER.debug("Subscription id = " + subscriptionId);
+         
+         // read the new rule wanted to be added
+        BufferedReader reader = request.getReader();
+        String endpointStr = "";
+        String line;
+
+        while ((line = reader.readLine()) != null) {
+            endpointStr += line;
+        } // while
+
+        reader.close();
+        
+        // Create a Gson for check JSON
+        Gson gson = new Gson();
+        OrionEndpoint endpoint;
+        
+        try {
+            endpoint = gson.fromJson(endpointStr, OrionEndpoint.class);
+        } catch (JsonSyntaxException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            response.getWriter().println("{\"success\":\"false\","
+                    + "\"error\":\"Parse error, malformed Json. Check it for errors.\"");
+            LOGGER.error("Parse error, malformed Json. Check it for errors.\"");
+            return;
+        } // try catch
+                       
+        // check if the endpoint are valid
```

Comment:
These checks for OrionEndpoint did not exist yet? I mean, the OrionEndpoint has already been used, thus these checks must be in the code somewhere. So, I think these checks should be within a function and be re-used.